File: leetcode/2236_Root_Equals_Sum_of_Children.cpp
#include<bits/stdc++.h>
using namespace std;
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 };
 
class Solution {
public:
    bool checkTree(TreeNode* root) {
        int child = root->left->val + root->right->val;
        return child == root->val;
    }
};
int main(){
    Solution s;
    TreeNode* root = new TreeNode(10);
    root->left = new TreeNode(4);
    root->right = new TreeNode(6);
    bool ans = s.checkTree(root);
    cout << (ans ? "true" : "false") << endl;
}

File: leetcode/2236_Root_Equals_Sum_of_Children.java
import java.util.*;
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}
class Solution {
    public boolean checkTree(TreeNode root) {
        return root.val == (root.left.val + root.right.val);
    }
}
class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
        TreeNode root = new TreeNode(10, new TreeNode(4), new TreeNode(6));
        System.out.println(solution.checkTree(root)); // Output: true
    }
}